python/ray/tests/test_failure_4.py

```diff
@@ -433,5 +434,39 @@ def ready(self):
     ray.get(a.ready.remote())
 
 
+def test_worker_start_timeout(monkeypatch, ray_start_cluster):
+    # This test is to make sure
+    #   1. when worker failed to register, raylet will print useful log
+    #   2. raylet will kill hanging worker
+    with monkeypatch.context() as m:
+        # delay internal kv for 5s
+        # this delay will make worker start slow
+        m.setenv(
+            "RAY_testing_asio_delay_us",
+            "InternalKVGcsService.grpc_server.InternalKVGet"
+            "=3000000:3000000")
```

Comment:
Change this to something higher like 999999999 to make it clear?

Reply:
@ericl I'm reverting this back. It looks like internal kv is not just used in worker start, but also driver starts. So make it 9999999... will make it timeout. I think here we need better control for this, but for now, let me just make it 2 seconds.